controllers/controllers/resource/template.go

```diff
@@ -159,7 +166,146 @@ func (r *VsphereTemplate) TemplateResources(ctx context.Context, eksaCluster *an
 		values["eksaVspherePassword"] = string(passwordBytes)
 	}
 
-	return generateTemplateResources(templateBuilder, clusterSpec, workloadTemplateNames, kubeadmconfigTemplateNames, cpOpt)
+	return generateTemplateResources(templateBuilder, clusterSpec, workloadTemplateNames, kubeadmconfigTemplateNames, cpOpt, nil)
+}
+
+func (r *CloudStackTemplate) TemplateResources(ctx context.Context, eksaCluster *anywherev1.Cluster, clusterSpec *cluster.Spec, csdc anywherev1.CloudStackDatacenterConfig, cpCsmc, etcdCsmc anywherev1.CloudStackMachineConfig, workerCsmcs map[string]anywherev1.CloudStackMachineConfig) ([]*unstructured.Unstructured, error) {
+	if len(workerCsmcs) > 1 {
+		return nil, fmt.Errorf("multiple worker node group configuration is not supported in CloudStack provider")
+	}
+	workerNodeGroupMachineSpecs := make(map[string]anywherev1.CloudStackMachineConfigSpec, len(workerCsmcs))
+	for _, wnConfig := range clusterSpec.Cluster.Spec.WorkerNodeGroupConfigurations {
+		workerNodeGroupMachineSpecs[wnConfig.MachineGroupRef.Name] = workerCsmcs[wnConfig.MachineGroupRef.Name].Spec
+	}
+	// control plane and etcd updates are prohibited in controller so those specs should not change
```

Comment:
I think that's something we want to support. What part of this PR doesn't support GitOps?

Reply:
Managing a workload cluster with GitOps, not just management cluster. With workload cluster, we can modify any part of the cluster spec like the CLI.

Reply:
I'm not sure I fully understand the problem. Do you think you could create a GH issue so that we can track it? Or is there an example in vsphere, for example, that we could reuse?

Reply:
We have an annotation that we set on workload clusters to signify what to do in the reconciliation. So we have these used in the webhooks and controllers. Some parts might be reusable (especially if it is related to the cluster). Then it's just a matter of handling changing the template names, which I think is already handled by this code.

machine config webhook: https://github.com/aws/eks-anywhere/blob/main/pkg/api/v1alpha1/vspheremachineconfig_webhook.go#L80

template code: https://github.com/aws/eks-anywhere/blob/918bbd2e19c662ce70e13695027a5fb2d6ed8b73/controllers/controllers/resource/template.go#L69

Reply:
Looking into this, it seems like a decent number of changes are required, so I'm inclined to hold off on implementation and create an issue instead. The current code supports multi-cluster creation/management which should suffice for the time being

Reply:
I think this shouldn't actually require a crazy amount of changes? It should just be checking for those annotations in those resources like in the machine config webhook above right?

Reply:
Will add a check for the annotations in the webhook